gcc-c++17.cmake

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2013, Ruslan Baratov
+# Copyright (c) 2017, Richard Hodges
+# All rights reserved.
+
+if(DEFINED POLLY_GCC_CXX17_CMAKE)
```

Comment:
Please change the name of the toolchain to `gcc-cxx17` similar to existent one.

Reply:
I'll revoke this pull request and submit individual ones